mem/kv_mem.go

```diff
@@ -220,19 +241,29 @@ func (kv *memKV) Capabilities() int {
 	return kvdb.KVCapabilityOrderedUpdates
 }
 
-func (kv *memKV) get(key string) (*kvdb.KVPair, error) {
+// get returns the value and creates a copy if specified
+func (kv *memKV) get(key string, copy bool) (*memKVPair, error) {
 	key = kv.domain + key
 	v, ok := kv.m[key]
 	if !ok {
 		return nil, kvdb.ErrNotFound
 	}
+	if kv.noByte && copy {
```

Comment:
this code is repeated at many places. It may be more clear to define methods on memKVPair like:
create()
copy()
get() will always return memKVPair as it exists, Enumerate/GetVal can then just do .copy() on returned memKVPair, should reduce code duplication.